genesis/sensors/data_handlers.py

```diff
@@ -220,22 +247,31 @@ class NPZFileWriter(DataHandler):
     def __init__(self, filename: str):
         assert filename.endswith(".npz"), "NPZ output must be an .npz file"
         self.filename = filename
-        self.all_data: list[np.ndarray] = []
+        self.all_data: dict[str, list] = defaultdict(list)
 
     def initialize(self):
         self.all_data.clear()
 
-    def process(self, data):
-        self.all_data.append(data)
+    def process(self, data, cur_time):
+        self.all_data["timestamp"].append(cur_time)
+        if isinstance(data, dict):
+            for key, value in data.items():
+                if isinstance(value, torch.Tensor):
+                    value = tensor_to_array(value)
+                self.all_data[key].append(value)
+        else:
+            self.all_data["data"].append(tensor_to_array(data))
 
     def cleanup(self):
         gs.logger.info(f'Saving data to "~<{self.filename}>~"...')
+        saved_data = {k: np.array(v) for k, v in self.all_data.items()}
```

Comment:
Are you sure converting to ndarray is necessary? Usually numpy handles this internally.